.github/workflows/test.yaml

```diff
@@ -60,7 +60,7 @@ jobs:
           # end to end functional test usage for non-distributed trial runs.
           # When updating the minimum Python version here, also update the
           # `python_requires` from `setup.cfg`.
-          - python-version: '3.7.0'
+          - python-version: '3.7.1'
```

Comment:
What was the rationale for this, out of interest?

Reply:
AFAIK, there is no python 3.7.0 available on GitHub Actions.

Reply:
Thanks, that makes sense. I'm not sure this means that Twisted itself needs to depend on Python >= 3.7.1 though? (See the change to setup.cfg in this commit)

Reply:
(For context, we found ourselves unable to create a Poetry lockfile when running against Twisted trunk: see https://github.com/matrix-org/synapse/issues/12575#issuecomment-1112020359)

Reply:
ok. you can send a PR for 3.7.0 and I can approve it.

previously we had `python_requires = >=3.6.7` ... so it wasn't 3.6.0 ... so we went with the minium GHA could offer.